src/main/java/com/mongodb/hibernate/internal/translate/AbstractMqlTranslator.java

```diff
@@ -392,16 +395,9 @@ private AstProjectStage createProjectStage(SelectClause selectClause) {
 
     @Override
     public void visitFromClause(FromClause fromClause) {
-        if (fromClause.getRoots().size() != 1) {
-            throw new FeatureNotSupportedException();
-        }
+        checkFromClauseSupportability(fromClause);
         var tableGroup = fromClause.getRoots().get(0);
-
-        if (!(tableGroup.getModelPart() instanceof EntityPersister entityPersister)
-                || entityPersister.getQuerySpaces().length != 1) {
-            throw new FeatureNotSupportedException();
-        }
-
+        var entityPersister = (EntityPersister) tableGroup.getModelPart();
         affectedTableNames.add(((String[]) entityPersister.getQuerySpaces())[0]);
```

Comment:
the above `visitFromClause` will only be invoked for `selection query`, not for `mutation query`, though some common validation logic applies to both. For that reason, the logic is centralized in a new `checkFromClauseSupportability()` method